### ISSUE TITLE
ReactJS and LaTeX

### ISSUE THREAD

**Issue by rtdavis22** [mathjax/MathJax#2563]
**Is your feature request related to a problem? Please describe.**
I'd like to write LaTeX in React components (with ReactJS). Something like this:
```
class Widget extends React.Component {
  render() {
    return <span>\(a_{ij}\)</span>;
  }
}
```
Unfortunately, this results in an error because of the braces:
```
'ij' is not defined  no-undef
```
However, the braces can be escaped:
```
<span>\(a_{'{'}ij{'}'}\)</span>
```
This produces the right result but is obviously very ugly.
This also works but is also pretty ugly:
```
<span dangerouslySetInnerHTML={{__html: '\\(a_{ij}\\)'}} />;
```
This works as well:
```
<span>\(a_&#123;ij&#125;\)</span>
```

**Describe the solution you'd like**
I'd like there to be a clean way of writing LaTeX in React components. 

**Describe alternatives you've considered**
I've considered implementing a preprocessor that searches for all braces ("{" and "}") inside LaTeX math delimiters and escapes them as shown above. I'd prefer a solution that doesn't involve adding a step to the build process.

Something else I've considered is a component that surrounds its children with braces:
```
class B extends React.Component {
  render() {
    return ['{', this.props.children, '}'];
  }
}

<span>\(a_<B>ij</B>\)</span>
```

**Additional context**

**Comment by tumidi** [comment]
Using MathJax with React is not straight-forward. From the [React documentation](https://reactjs.org/docs/integrating-with-other-libraries.html#integrating-with-dom-manipulation-plugins):

> React is unaware of changes made to the DOM outside of React. It determines updates based on its own internal representation, and if the same DOM nodes are manipulated by another library, React gets confused and has no way to recover.

Because this is a common problem for everyone who wants to use MathJax with React, I created a library that solves this problem in an elegant, performant but also customizable way:

https://github.com/innodoc/react-mathjax-node

In a nutshell, the library never lets MathJax do any DOM manipulation by itself (which makes React happy). It's a set of very few easy to use React components. There's no need to work with any math delimiters (`\(`, `\)`, etc.). You just hand the TeX code to the component and that's it.

It works by letting MathJax do the heavy lifting of generating DOM elements from TeX code. The whole DOM manipulation of inserting and removing these nodes is left to React. Furthermore you can easily add custom callbacks and customize the MathJax configuration. Also, loading, initializing and managing the MathJax library is done automatically in the background.

**Comment by dpvc** [comment]
Closed as part of a review of old issues.
